### PR TITLE
Add `resolve` option

### LINKED ISSUE #2: Dont call path.resolve
I need to construct file URLs for a Linux system in a script running on Windows. When I do this:

``` js
fileUrl('/test/test.txt')
```

results in 

```
file://c:/test/text.txt
```

Which is totally wrong of course. The C drive letter comes from path.resolve, which resolves relative to the cwd.

Please either remove this or add a parameter to turn it off

### PR DESCRIPTION
**Pull request by felixfbecker**
This fixes #2 without any breaking changes.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 
-module.exports = function (str) {
+module.exports = function (str, resolve) {
 	if (typeof str !== 'string') {
 		throw new Error('Expected a string');
 	}
+	
+	var pathName = str;
+	
+	if (resolve || typeof resolve === 'undefined') {
```
Comment (SamVerschueren) [inline index.js]: `if (resolve !== false)` might be better.

**Comment by felixfbecker** [comment]
Btw, are you aware that your tests don't pass on Windows because of the issue I described in #2?

```
1)  should convert path to file url:

      AssertionError: 'file:///C:/Users/sindresorhus/dev/te%5Est.jpg' === 'file:///Users/sindresorhus/dev/te%5Est.jpg'
      + expected - actual

      -file:///C:/Users/sindresorhus/dev/te%5Est.jpg
      +file:///Users/sindresorhus/dev/te%5Est.jpg
```

It is because of `path.resolve()` ;)

File: index.js (hunk 1 above)
Comment (felixfbecker) [inline index.js]: changed it

**Comment by felixfbecker** [comment]
I added tests and fixed tests on Windows

File: test.js [hunk 2]
```diff
@@ -4,5 +4,13 @@ var fileUrl = require('./');
 
 it('should convert path to file url', function () {
 	assert(/file:\/\/\/.*\/test\.jpg/.test(fileUrl('test.jpg')));
-	assert.strictEqual(fileUrl('/Users/sindresorhus/dev/te^st.jpg'), 'file:///Users/sindresorhus/dev/te%5Est.jpg');
+	if (process.platform === 'win32') {
```
Comment (sindresorhus) [inline test.js:7]: This is not really the way to fix it. Some users might have a different drive than `C:`. Instead, just do two check, one that check the it starts with `file:////` and one that it ends in `te%5Est.jpg`.

File: readme.md [hunk 3]
```diff
@@ -20,6 +20,10 @@ fileUrl('unicorn.jpg');
 
 fileUrl('/Users/pony/pics/unicorn.jpg');
 //=> 'file:///Users/pony/pics/unicorn.jpg'
+
+// passing false as second argument will not call path.resolve() on the path
+fileUrl('unicorn.jpg', false);
```
Comment (sindresorhus) [inline readme.md]: Should be an options object, not a [boolean trap](http://ariya.ofilabs.com/2011/08/hall-of-api-shame-boolean-trap.html).

File: test.js (hunk 2 above)
Comment (felixfbecker) [inline test.js]: Well, at the moment you are assuming everyone has this repo cloned in a user directory named `sindresorhus`. I think that is a much bigger assumption ;)

Comment (felixfbecker) [inline test.js:7]: It doesn't matter what drive the user runs this on as I pass an absolute path that already contains a drive letter. In fact, I just ran the tests on my `D:` drive and they pass just fine. It is just about passing a Windows-compatible path.

**Comment by felixfbecker** [comment]
I changed it to an options object. The tests are fine how they are, see my comment.

**Comment by felixfbecker** [comment]
great :)

**Pull request closed without merging**